stdlib/template/configure.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python3.6
+# -*- coding: utf-8 -*-
+"""Provides a small, partial template that wraps the ``configure`` command."""
+
+import os
+import stdlib
+from typing import List
+
+
+def get_dir_flags() -> List[str]:
+    """Build a list of directory flags used by the :py:func:`.configure` template.
+
+    The return value is a list of directory flags that indicates where the build software should be located
+    in the filesystem.
+
+    For example, it will provide flags like::
+
+        --prefix='/usr'
+        --bindir='/usr/bin'
+
+    etc.
+
+    :returns: A list of flags that can be given to a configure script.
+    """
+    return [
+        '--prefix="/usr"',
+        '--bindir="/usr/bin"',
+        '--sbindir="/usr/bin"',
+        '--libdir="/usr/lib64"',
+        '--libexecdir="/usr/libexec"',
+        '--includedir="/usr/include"',
+        '--datarootdir="/usr/share"',
+        '--datadir="/usr/share"',
+        '--infodir="/usr/share/info"',
+        '--mandir="/usr/share/man"',
+        '--sysconfdir="/etc"',
+        '--localstatedir="/var"',
+    ]
+
+
+def get_feature_flags() -> List[str]:
+    """Build a list of feature flags used by the :py:func:`.configure` template.
+
+    The return value is a list of feature flags that enable parts of the target software based
+    on the available dependencies.
+
+    For example, it will provide flags like::
+
+        --with-curses
+        --with-pthread
+
+    etc.
+
+    :returns: A list of flags that can be given to a configure script.
+    """
+    return [
+        '--with-curses',
+        '--with-cursesw',
+        '--with-ncurses',
+        '--with-ncursesw',
+        '--with-pthreads',
+        '--with-threads',
+        '--with-readline',
+        '--with-history',
+    ]
+
+
+def get_flags() -> List[str]:
+    """Build a list of system-wide flags used by the :py:func:`.configure` template.
+
+    :returns: The merged output of :py:func:`.get_dir_flags` and :py:func:`.get_feature_flags`.
+    """
+    return get_dir_flags() + get_feature_flags()
+
+
+def configure(
+    *flags: str,
+    system_flags: bool = True,
+    binary: str = './configure',
+):
+    """Run ``./configure`` with a specific set of arguments.
+
+    The flags given to configure are both the return value of :py:func:`.get_flags()` and the given ``flags``.
+
+    Some configure script abort on unknown flags. If ``system_flags`` is ``False``, the default system flags
```

Comment:
```suggestion
    Some configure script aborts on unknown flags. If ``system_flags`` is ``False``, the default system flags
```
Also did you meant to say `Some configure script`, and not something like `this configure script` or `the configure script`?

Reply:
I think he meant to use a plural but forgot the "s" for "script"